api/src/main/java/org/openmrs/module/openconceptlab/OpenConceptLabActivator.java

```diff
@@ -93,6 +97,25 @@ public void contextRefreshed() {
 		log.info("Open Concept Lab Module refreshed");
 	}
 
+	/**
+	 * If OpenMRS is unexpectedly shutdown while an OCL Import is in progress, the import record in the database
+	 * can be left in an "in progress" state, even though the import is no longer actually running.
+	 * In this situation, we want to ensure these imports are marked as failed at the next startup, so that
+	 * subsequent import attempts can succeed without errors that imports are already in progress
+	 */
+	protected void markInProgressImportsAsFailed() {
+		ImportService importService = Context.getService(ImportService.class);
+		List<Import> inProgressImports = importService.getInProgressImports();
+		if (!inProgressImports.isEmpty()) {
+			log.warn("Found " + inProgressImports.size() + " in progress imports at startup");
+			for (Import inProgressImport : inProgressImports) {
+				log.warn("Updating Import#" + inProgressImport.getImportId() + " as failed");
+				importService.failImport(inProgressImport, "System restarted during import");
```

Comment:
super nit, but this also could happen if the context is refreshed (ie someone manually loads a module via the old UI) while an import is rerunning.  Just wondering if the message should be "Context refreshed and/or system restarted during import."

Reply:
@mogoodrich thanks I updated the message.